core/src/main/java/org/apache/calcite/sql/SqlUnnestOperator.java

```diff
@@ -67,7 +71,19 @@ public SqlUnnestOperator(boolean withOrdinality) {
       RelDataType type = opBinding.getOperandType(operand);
       if (type.isStruct()) {
         type = type.getFieldList().get(0).getType();
+      } else if (type.getSqlTypeName() == SqlTypeName.ANY) {
```

Comment:
It is interesting that the output of Unnest is being treated as ANY even though it is technically a struct, so the first 'if' check should have matched.  Some explanation here would be useful.